package.json

```diff
@@ -1,6 +1,6 @@
 {
   "name": "multinet",
-  "version": "0.18.0",
+  "version": "0.0.0-push-release",
```

Comment:
Should this be changed back to `0.18.0`?

Reply:
Well, my goal is for us to never have to update this value in any future pull requests, so it sort of doesn't matter what value it is... other than signalling that contributors shouldn't change it. The idea is that in CI, when the NPM publishing process happens, a local commit is created with the new version number (which updates this field in the local checkout), and that state is used to publish the package. Since the version field change is the *only* change, we don't actually need to have that commit in our history.

In lieu of using the version field, the best way to know what version you're on will be to use `git describe`, which is something we may want to mention in documentation or something.

Reply:
It seems like we're going against best practices not keeping this in sync with what's on npm, but I'm not sure if there's a sane way to achieve this, if we're also trying to make this work with pre-releases / deploy previews. As in, you could have part of CI be to update the version in master, but then I'm not sure how you would separate this from the other branches with tags that are just for previews. What do you think?

Reply:
If you're suggesting we could run a job on master that looks for the last version tag and updates the version field to match, that would be simple enough but I'm not sure what benefit it would achieve. It also has certain costs that would be annoying for us (for instance, if you merged several PRs in a row, you'd have CI pushing a bunch of commits to master in a way that would be out-of-sync with reality, or would cause conflicts).

For comparison (and de facto best practices), take a look at how [`semantic-release`](https://github.com/semantic-release/cli#what-it-does) works; in particular, step 4 of that process advises setting your package.json version field to `0.0.0-development` then letting CI manage the version field, which is essentially what I've done here.

Since semantic-release does it this way, I'd say we should too. It eliminates the need to maintain even more CI infrastructure just to attempt to get a version field mapped to NPM, which I don't think brings us any benefit anyway.